packages/storage/src/implementation/request.ts

```diff
@@ -46,15 +40,24 @@ export interface Request<T> {
   cancel(appDelete?: boolean): void;
 }
 
-class NetworkRequest<T> implements Request<T> {
-  private pendingConnection_: Connection | null = null;
+/**
+ * Handles network logic for all Storage Requests, including error reporting and
+ * retries with backoff.
+ *
+ * @param I - the type of the backend's network response (always `string`,
```

Comment:
Sounds like we should add a constraint to the generic parameter? e.g. `<I extends string | ArrayBuffer |  ReadableStream, O>`?